src/src/beneficiaries/beneficiaries.service.ts

```diff
@@ -2077,4 +2135,92 @@ export class BeneficiariesService {
 			data: result,
 		};
 	}
+
+	public async getAllDuplicatesUnderIp(id: number) {
+		const user = (await this.findOne(id)).data;
+		const sql = `
+			SELECT
+				bu.aadhar_no AS "aadhar_no",
+				COUNT(*) AS "count"
+			FROM
+				users bu
+			INNER JOIN
+				program_beneficiaries pb
+			ON
+				bu.id = pb.user_id
+			LEFT OUTER JOIN
+				users fu
+			ON
+				pb.facilitator_id = fu.id
+			LEFT OUTER JOIN
+				program_faciltators pf
+			ON
+				fu.id = pf.user_id
+			WHERE
+				pf.parent_ip = '${user?.program_users?.organisation_id}'
+			AND
+				bu.aadhar_no IS NOT NULL
+			GROUP BY
+				bu.aadhar_no
+			HAVING
+				COUNT(*) > 1
+			AND
+				COUNT(*) = (
+					SELECT
+						COUNT(*)
+					FROM
+						users bu2
+					INNER JOIN
+						program_beneficiaries pb2
+					ON
+						bu2.id = pb2.user_id
+					WHERE
+						bu2.aadhar_no = bu.aadhar_no
+				)
+			;
+		`;
+		const duplicateListArr = (
+			await this.hasuraServiceFromServices.getDataFromSQL(sql)
+		).result;
+		return this.hasuraServiceFromServices.getFormattedData(
+			duplicateListArr,
+		);
+	}
+
+	public async getAllDuplicatesUnderPo() {
```

Comment:
internal methods shall be private

Reply:
We won't be able to call it in controller of we make it private